mavros_extras/src/plugins/vision_speed_estimate.cpp

```diff
@@ -76,38 +78,53 @@ class VisionSpeedEstimatePlugin : public plugin::PluginBase {
 		vs.y = v.y();
 		vs.z = v.z();
 		// [[[end]]] (checksum: aee3cc9a73a2e736b7bc6c83ea93abdb)
+		ftf::covariance_to_mavlink(cov, vs.covariance);
 
 		UAS_FCU(m_uas)->send_message_ignore_drop(vs);
 	}
 
-	/**
-	 * @todo Suggest modification on PX4 firmware to MAVLINK VISION_SPEED_ESTIMATE
-	 * msg name, which should be called instead VISION_VELOCITY_ESTIMATE
-	 */
-
 	/* -*- mid-level helpers -*- */
 
 	/**
 	 * @brief Send vision speed estimate to FCU velocity controller
 	 */
-	void send_vision_speed(const geometry_msgs::Vector3 &vel_enu, const ros::Time &stamp)
+	void send_vision_speed(const ros::Time &stamp, const geometry_msgs::Vector3 &vel_enu, const mavros::ftf::Covariance3d &cov)
 	{
 		Eigen::Vector3d vel_;
 		tf::vectorMsgToEigen(vel_enu, vel_);
-		//Transform from ENU to NED frame
-		auto vel = ftf::transform_frame_enu_ned(vel_);
+		// Transform from ENU to NED frame
+		auto vel_ned = ftf::transform_frame_enu_ned(vel_);
+		auto cov_ned = ftf::transform_frame_enu_ned(cov);
 
-		vision_speed_estimate(stamp.toNSec() / 1000, vel);
+		vision_speed_estimate(stamp.toNSec() / 1000, vel_ned, cov_ned);
 	}
 
 	/* -*- callbacks -*- */
 
-	void vel_twist_cb(const geometry_msgs::TwistStamped::ConstPtr &req) {
-		send_vision_speed(req->twist.linear, req->header.stamp);
+	void vel_twist_with_cov_cb(const geometry_msgs::TwistStamped::ConstPtr &req) {
+		ftf::Covariance3d cov {};	// zero initialized
```

Comment:
Does mavlink propose some value for incorrect cov? Like [-1, 0, 0...] in ROS msgs.

Reply:
It doesn't. But should be a good idea to bring yes.